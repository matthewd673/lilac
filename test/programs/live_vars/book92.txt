[BB_00]
i32 $i = 1i32

[BB_01 ENTRY=[L0:] EXIT=[jnz %0 L1]]
i32 $a = 2i32
i32 $c = 3i32
i32 %0 = $a < $c

[BB_02]
i32 $b = 4i32
i32 $c = 5i32
i32 $d = 6i32

[BB_03 ENTRY=[L3:] EXIT=[jz %1 L0]]
i32 $y = $a + $b
i32 $z = $c + $d
i32 $i = $i + 1i32
i32 %1 = $i <= 100i32

[BB_04]
ret void

[BB_05 ENTRY=[L1:] EXIT=[jnz %2 L2]]
i32 $a = 7i32
i32 $d = 8i32
i32 %2 = $a <= $d

[BB_06]
i32 $d = 9i32

[BB_07 ENTRY=[L4:] EXIT=[jmp L3]]
i32 $b = 10i32

[BB_08 ENTRY=[L2:] EXIT=[jmp L4]]
i32 $c = 11i32
