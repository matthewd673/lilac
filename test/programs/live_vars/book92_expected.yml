- id: ENTRY
  GEN: []
  KILL: []
- id: BB_00
  GEN: []
  KILL:
    - i
- id: BB_01
  GEN: []
  KILL:
    - a
    - c
    - "%0" # not in book
- id: BB_02
  GEN: []
  KILL:
    - b
    - c
    - d
- id: BB_03
  GEN:
    - a
    - b
    - c
    - d
    - i
  KILL:
    - y
    - z
    - i
    - "%1" # not in book
- id: BB_04
  GEN: []
  KILL: []
- id: BB_05
  GEN: []
  KILL:
    - a
    - d
    - "%2" # not in book
- id: BB_06
  GEN: []
  KILL:
    - d
- id: BB_07
  GEN: []
  KILL:
    - b
- id: BB_08
  GEN: []
  KILL:
    - c
- id: EXIT
  GEN: []
  KILL: []
